src/ens/ens_fs.c

```diff
@@ -0,0 +1,103 @@
+#include <drivers/flash.h>
+#include <storage/flash_map.h>
+#include <string.h>
+#include <sys/crc.h>
+
+#include "ens_error.h"
+#include "ens_fs.h"
+
+#define SEED 42
+
+int ens_fs_init(ens_fs_t* fs, uint8_t id, uint64_t entry_size) {
```

Comment:
The entry_size here is not the real size of an entry as we include another byte for CRC and deletion flag?
This seems a bit confusing to me?

Reply:
Please also rename the id to flash_id
```suggestion
int ens_fs_init(ens_fs_t* fs, uint8_t flash_id, uint64_t entry_size) {
```